### PR TITLE
Unify framework integration examples

### PR DESCRIPTION
**Pull request by jacekbogdanski**
Closes https://github.com/ckeditor/ckeditor4/issues/3821
Closes https://github.com/ckeditor/ckeditor4/issues/4305

### CODE REVIEW

**Comment by f1ames** [comment]
Maybe we could also cover ckeditor/ckeditor4#4305 here?

**Comment by Dumluregn** [comment]
☝️  I was thinking about it too. Shouldn't create too much fuzz to this PR. The thing that stopped me was it's still in `Planning` 🙂

**Comment by jacekbogdanski** [comment]
> Maybe we could also cover ckeditor/ckeditor4#4305 here?

Done.

File: scripts/integrations/angular/src/app/binding/editor-binding.component.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+} from '@angular/core';
+
+@Component( {
+	selector: 'app-binding',
+	templateUrl: './editor-binding.component.html'
+} )
+export class EditorBindingComponent implements AfterViewInit {
+	@ViewChild( 'editor' ) editor;
+
+	public editorData = '<p>This is a CKEditor 4 WYSIWYG editor instance created with Angular.</p>'
```
Comment (Dumluregn) [inline scripts/integrations/angular/src/app/binding/editor-binding.component.ts:19]: Angular version of this component has a different (richer) toolbar than the React and Vue ones.

File: scripts/integrations/angular/src/app/binding/editor-binding.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
```
Comment (Dumluregn) [inline scripts/integrations/angular/src/app/binding/editor-binding.component.ts:2]: The years here should be `2003-2020`. This also needs to be fixed in several other angular and vue files.

**Review by Dumluregn — CHANGES_REQUESTED** [review]
All samples work correctly, well done 👍 A lot of work is already done, now let's proceed with some cleanup.

Besides the review comments:

- [x] Vue and React have the `CKEditor 4 WYSIWYG Editor` prefix in the `h1` tag, while Angular has just `CKEditor 4`. The latter seems to be more approachable, but I suppose the longer version was created with SEO in mind, so let's stand with that one and adjust the Angular one.

- [x] In Angular sample, in the first sentence we have:
  > The Angular 5+ integration

  I think that the version number here looks a bit strange. We have this information in the [guide](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_angular.html) and [npm readme](https://www.npmjs.com/package/ckeditor4-angular), so I'd remove it from here.
- [x] In the sentence
  > To use the CKEditor 4 classic editor in Angular, create a new <ckeditor /> tag.

  Let's change `tag` to component here to be consistent with React and Vue.
- [ ] In Vue sample (`vue.html` file) there is a reference to a non-existing file:
  > `<link rel="stylesheet" href="assets/vue/samples.css">`

  We should remove it as the styles in this sample are nicely encapsulated inside the components.
- [ ] In React sample the styles are scattered between the `react.html` and `assets/react/samples.css` files and some of them are overlapping. Let's move them all to the external file and audit them there.

**Comment by jacekbogdanski** [comment]
> In React sample the styles are scattered between the react.html and assets/react/samples.css files and some of them are overlapping. Let's move them all to the external file and audit them there.

Actually, these styles weren't loaded at all. It seems like custom style links are somehow removed probably via bundling mechanism. Due to that, I would be for keeping integration styles inside example files to keep it simple.

File: docs/sdk/examples/angular.html [hunk 3]
```diff
@@ -233,91 +203,102 @@ <h1>CKEditor 4 Angular Integration <a class="documentation" href="https://ckedit
...
 } )
 export class EventsComponent {
-	public isReadOnly = false;
-	public isHidden = false;
-	public isRemoved = false;
-	public componentEvents: ComponentEvent[] = [];
+	public events: ComponentEvent[] = [];
+
+	public editorConfig = {
+		toolbar: [
```
Comment (Dumluregn) [inline docs/sdk/examples/angular.html:251]: Please remove the custom toolbar config from the `Configuration` samples for all frameworks as you did it for the `Two-way Data Binding` ones. Also code snippets need to be adjusted afterwards.

File: docs/sdk/examples/angular.html [hunk 4]
```diff
@@ -51,7 +51,6 @@
 				outline: #2ba2f1 auto 5px;
 			}
 
-
 			.angular-nav__menu {
```
Comment (Dumluregn) [inline docs/sdk/examples/angular.html:54]: Please add `padding: 0` here to remove small padding that is inherited from somewhere and isn't present in other frameworks samples.

**Review by Dumluregn — CHANGES_REQUESTED** [review]
> Actually, these styles weren't loaded at all. It seems like custom style links are somehow removed probably via bundling mechanism.

That's a shame, because I hoped that maybe at some point we were able to extract the common styles (like the tabs or event log styling) from these 3 samples to a common stylesheet to make the future changes easier 😞 

---------------------

I see our code snippets with source code are a little inconsistent with how much do they reflect the docs demo:

| Framework | sample | HTML |
| --- | --- | --- |
| Angular | Events | ✅  |
| Angular | 2-way DB | ✅ |
| React | Events | ❎ |
| React | 2-way DB | ✅ |
| Vue | Events | ❎ |
| Vue | 2-way DB | ❎ |

(By `HTML` in the table I mean e.g. event log in `Events` sample and the box with rendered content in `2-way DB` samples).

I think it would be nice to make it consistent at least at the integration level, so if you find it too time consuming to expand the Vue samples, please just take care of the React `Events` sample.

File: docs/sdk/examples/angular.html (hunk 4 above)
Comment (jacekbogdanski) [reply docs/sdk/examples/angular.html:54]: `padding: 0` is already here, the selector is not specific enough.

File: docs/sdk/examples/angular.html (hunk 3 above)
Comment (jacekbogdanski) [reply docs/sdk/examples/angular.html:251]: This sample shows how to apply a custom config for the editor, that's the reason for the bookmark name `Configuration`.

**Comment by jacekbogdanski** [comment]
> That's a shame, because I hoped that maybe at some point we were able to extract the common styles (like the tabs or event log styling) from these 3 samples to a common stylesheet to make the future changes easier disappointed

No shame at all. These are 3 different frameworks which may produce different HTML output. Keeping a single CSS stylesheet for them may provide more maintenance issues than separating them, because changing one CSS rule may actually break it for different examples (due to differences how frameworks produce HTML). A small difference between samples is better than accidentally braking one sample when changing another.

**Comment by jacekbogdanski** [comment]
Note that JSX rendering doesn't look good because of https://github.com/cksource/umberto/issues/606 upstream.

File: docs/sdk/examples/angular.html (hunk 3 above)
Comment (Dumluregn) [reply docs/sdk/examples/angular.html:251]: Right, it makes sense 🙈

**Review by Dumluregn — APPROVED** [review]
LGTM! 🎉

**Pull request merged**